confmap/provider.go

```diff
@@ -138,9 +140,14 @@ func NewRetrievedFromYAML(yamlBytes []byte, opts ...RetrievedOption) (*Retrieved
 		return nil, err
 	}
 
-	switch v := rawConf.(type) {
+	switch rawConf.(type) {
 	case string:
-		opts = append(opts, withStringRepresentation(v))
+		if featuregates.UseUnifiedEnvVarExpansionRules.IsEnabled() {
+			// If rawConf is a string, we override it with the raw yamlBytes.
+			// This is the original ${ENV} expansion behavior.
+			rawConf = string(yamlBytes)
+		}
+		opts = append(opts, withStringRepresentation(string(yamlBytes)))
```

Comment:
This changes the behavior when the feature gate is disabled.  Before it would set the string representation to be the yaml parsed version of the string (so `\n` is removed.

Reply:
Yup, but the string representation is only used under the strict types feature gate, so it's also under a feature gate